runtime/ms-rest-azure/lib/filters/rpRegistrationFilter.js

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information. 
+
+'use strict';
+
+var request = require('request');
+var utils = require('../utils');
+

```

Comment:
Please document this to the extent possible - 
1. what it does, why is it here
2. how it is different from other filters, in the sense that it can send out requests etc.